python/hsfs/constructor/query.py

```diff
@@ -221,21 +221,18 @@ def as_of(self, wallclock_time=None, exclude_until=None):
             when calling the `insert()` method.
 
         # Arguments
-            wallclock_time: Datetime string. The String should be formatted in one of the
+            wallclock_time: timestamp, Datetime object or string. The String should be formatted in one of the
                 following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, or `%Y%m%d%H%M%S`.
-            exclude_until: Datetime string. The String should be formatted in one of the
+            exclude_until: timestamp, Datetime object or string. The String should be formatted in one of the
                 following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, or `%Y%m%d%H%M%S`.
 
         # Returns
             `Query`. The query object with the applied time travel condition.
         """
-        wallclock_timestamp = (
-            util.get_timestamp_from_date_string(wallclock_time)
-            if wallclock_time
-            else None
-        )
+        wallclock_timestamp = util.convert_event_time_to_timestamp(wallclock_time)
+
         exclude_until_timestamp = (
-            util.get_timestamp_from_date_string(exclude_until)
+            util.convert_event_time_to_timestamp(exclude_until)
             if exclude_until
```

Comment:
We can exclude the null check here, because this is done in `util.convert_event_time_to_timestamp` now as well